### PR TITLE
Adding the `onArmorTick` item method

### PR DESCRIPTION
**Pull request by ZestyBlaze**
Hi hopefully this is okay but I've taken a crack at writing a version of the `onArmorTick` method from Forge's Item class, allowing players to easily add on-tick functionality to armor. I've tested it with my own custom armor I made and it should all work perfectly fine. I believe it's all optimised so it shouldn't cause a problem. If there is any additions or changes that need making, please feel free to tell me and I'll make the changes asap. Hope you enjoy :)

### CODE REVIEW

**Comment by TropheusJ** [comment]
Can you move it to a separate `ArmorTickListeningItem` interface? Extensions should be minimized.

**Comment by AlphaMode** [comment]
Why did you put the mixin in Player#tick? Forge injects into Inventory#tick

**Comment by ZestyBlaze** [comment]
Oh so it is, bizarre, I couldn't find that when I was looking, I'll make those changes rq and push them

**Comment by ZestyBlaze** [comment]
There we go that should hopefully fix the changes asked from

File: base/src/main/java/io/github/fabricators_of_create/porting_lib/item/ArmorTickListeningItem.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package io.github.fabricators_of_create.porting_lib.item;
+
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+
+public interface ArmorTickListeningItem {
+	default void onArmorTick(ItemStack stack, Level level, Player player) {
```
Comment (TropheusJ) [inline base/src/main/java/io/github/fabricators_of_create/porting_lib/item/ArmorTickListeningItem.java]: shouldn't be default

File: base/src/main/java/io/github/fabricators_of_create/porting_lib/mixin/common/PlayerMixin.java [hunk 2]
```diff
@@ -38,6 +40,9 @@ public abstract class PlayerMixin extends LivingEntity {
 	@Shadow
 	public abstract void disableShield(boolean sprinting);
 
+	@Shadow
+	public abstract Iterable<ItemStack> getArmorSlots();
```
Comment (TropheusJ) [inline base/src/main/java/io/github/fabricators_of_create/porting_lib/mixin/common/PlayerMixin.java]: remove now that it's unused

**Review by TropheusJ — COMMENTED** [review]
almost

File: base/src/main/java/io/github/fabricators_of_create/porting_lib/item/ArmorTickListeningItem.java (hunk 1 above)
Comment (TropheusJ) [reply base/src/main/java/io/github/fabricators_of_create/porting_lib/item/ArmorTickListeningItem.java]: and the javadoc is gone

**Comment by ZestyBlaze** [comment]
There we go :)

**Pull request merged**